### PR TITLE
Create Enumerable

### PR DESCRIPTION
**Pull request by haroldmud**
## In  this branch we did the following: :point_down: 

- Created a class `MyList` that has an instance variable `@list`.
- Implemented a method `#each` in `myList` that yields successive members of `@list` and uses the MyEnumerable module.
- Created a module `MyEnumerable` that implements the given methods(ALL, ANY, FILTER).
- Verified the solutions.
  - RESULT
Screenshot from 2022-07-15 11-54-02

### CODE REVIEW

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+class MyList
+  include MyEnumerable
+
+  def initialize(list)
+    @list = list
+  end
+
+  def each(&block)
+    @list.each(&block)
+  end
```
Comment (keroloussamy) [inline my_list.rb]: - [ ] Kindly in each method you need to implement its login not directly use each() method inside it. Note you can use the yield keyword to do that.

**Review by keroloussamy — CHANGES_REQUESTED** [review]
Hi @haroldmud, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Check the comments under the review._

- [ ] x

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zhadier — CHANGES_REQUESTED** [review]
# Hi @haroldmud  & @mandelatuks  👋  

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet. 🌤️ 



 ## Highlights

✔️ No linter errors
✔️ Githflow was setup correctly
✔️ Professional pull request title and description

 ## Required Changes ♻️

- [ ] As the previous reviewer requested, you are not supposed to use the inbuilt `each` method for arrays. Instead you can use a loop to create your own each method from scratch and use yield inside to assign functionality. For example you can do sth like
``` ruby
  def each
    i = 0
    while i < @list.length
       # Your functionality #
    end
  end
```

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question @zhadier so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zhadier — APPROVED** [review]
## Hi @haroldmud & @mandelatuks  👋  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 

### Keep Rocking!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mandelatuks** [comment]
@zhadier 
thank you for your guidance and it helped understand more now 
cheers and happy coding

**Pull request merged**